### PR TITLE
API: add nvim_namespace_clear_hl

### PR DESCRIPTION
**Pull request by shadmansaleh**
Currently it's not possible to clear highlight groups   from namespaces . This pr adds that .
It can clear specific highlight groups if highligh group names are given else can clear entire namespace .

### CODE REVIEW

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -233,6 +233,32 @@ void nvim_set_hl(Integer ns_id, String name, Dictionary val, Error *err)
...
+{
+  if (highlight_groups.size == 0) {
+    // empty name clear entire ns_id
+    ns_clear_hl((NS)ns_id);
+  } else {
+    // clear hlgroups in names from ns
+    String name = STRING_INIT;
+    for (size_t i = 0; i < highlight_groups.size; i++) {
+      if (highlight_groups.items[i].type == kObjectTypeString) {
+        name = copy_string(highlight_groups.items[i].data.string);
```
Comment (bfredl) [inline src/nvim/api/vim.c:253]: no need to copy the string (this is probably the memory leak)

Comment (shadmansaleh) [reply src/nvim/api/vim.c]: Thanks . It worked :) You can give a review if you have time .

**Comment by shadmansaleh** [comment]
I think I need to call highlight_clear() from [here](https://github.com/neovim/neovim/blob/cd3e74201a8fcbe5ba04000a57e05f102c04496c/src/nvim/syntax.c#L7106-7128) to clear the syntax but it's a static function is there any reason to keep it static . Can I remove the static so I can call it from highlight.c ?

**Comment by bfredl** [comment]
sure, removing `static` is typically fine.

**Comment by shadmansaleh** [comment]
Thanks . Should I be concerned about windows CI failing ? It's failing for something unrelated and tests in test/functional/api.vim_spec.lua and test/functional/ui/output_spec.lua passes in my device . I'm really not sure what's wrong with windows :P .

**Comment by clason** [comment]
Drive-by bikeshedding: `nvim_del_hl` for consistency with `get`, `set` (which also take a namespace argument without making it explicit in the name)?

**Comment by clason** [comment]
superseded by #17421

**Pull request closed without merging**